src/duration/std.rs

```diff
@@ -21,15 +21,16 @@ impl From<Duration> for std::time::Duration {
     /// 1. If the duration is negative, this will return a std::time::Duration::ZERO.
     /// 2. If the duration larger than the MAX duration, this will return std::time::Duration::MAX
     fn from(hf_duration: Duration) -> Self {
-        let (sign, days, hours, minutes, seconds, milli, us, nano) = hf_duration.decompose();
-        if sign < 0 {
-            std::time::Duration::ZERO
-        } else {
-            // Build the seconds separately from the nanos.
-            let above_ns_f64: f64 =
-                Duration::compose(sign, days, hours, minutes, seconds, milli, us, 0).to_seconds();
-            std::time::Duration::new(above_ns_f64 as u64, nano as u32)
-        }
+        const NANOS_PER_SEC: u128 = std::time::Duration::from_secs(1).as_nanos();
```

Comment:
I think a simpler way to implement this functionality is by using the [`to_unit`](https://docs.rs/hifitime/latest/hifitime/struct.Duration.html#method.to_unit) function on the `hf_duration`.

I haven't tested the following, so consider this a draft.

```rust
let seconds = hf_duration.to_seconds().floor() as u64; // Somehow check that this won't fail
let subsec_nanos = (hf_duration - Unit::Second * seconds).to_unit(Unit::Nanoseconds).floor() as u32; // Also check that this won't fail, probably using try_from as you initially recommended

std::time::Duration(seconds, subsec_nanos)
```

Reply:
There's a better way to handle the `subsec_nanos` than what I wrote above. After the subtraction, we know that the duration has zero centuries. So, we can safely ignore that part and operate directly on nanoseconds.

```rust
use crate::NANOSECONDS_PER_SECOND; // https://docs.rs/hifitime/latest/hifitime/constant.NANOSECONDS_PER_SECOND.html
let subsec_nanos: u64 = (hf_duration - Unit::Second * seconds).nanoseconds / NANOSECONDS_PER_SECOND;
// Finally convert this to u32 safely.
```